pkg/deployment/resources/pod_creator.go

```diff
@@ -662,11 +662,15 @@ func (r *Resources) createPodForMember(spec api.DeploymentSpec, memberID string,
 		engine := spec.GetStorageEngine().AsArangoArgument()
 		requireUUID := group == api.ServerGroupDBServers && m.IsInitialized
 		finalizers := r.createPodFinalizers(group)
-		if err := k8sutil.CreateArangodPod(kubecli, spec.IsDevelopment(), apiObject, role, m.ID, m.PodName, m.PersistentVolumeClaimName, imageInfo.ImageID, lifecycleImage, alpineImage, spec.GetImagePullPolicy(),
+		if err = k8sutil.CreateArangodPod(kubecli, spec.IsDevelopment(), apiObject, role, m.ID, m.PodName, m.PersistentVolumeClaimName, imageInfo.ImageID, lifecycleImage, alpineImage, spec.GetImagePullPolicy(),
 			engine, requireUUID, terminationGracePeriod, args, env, finalizers, livenessProbe, readinessProbe, tolerations, serviceAccountName, tlsKeyfileSecretName, rocksdbEncryptionSecretName,
 			clusterJWTSecretName, groupSpec.GetNodeSelector(), groupSpec.PriorityClassName, groupSpec.Resources, exporter, groupSpec.GetSidecars(), groupSpec.VolumeClaimTemplate); err != nil {
 			return maskAny(err)
 		}
+
+		m.ArangoVersion = status.CurrentImage.ArangoDBVersion
```

Comment:
Maybe we should store the _image_, not the version. You can always convert Image->Version.

Reply:
Maybe this should be generic code. You could move this up to the place where the image info is selected. Doing so would also set the image for Syncer Pods.

Reply:
- one too many update ermoved
- the image id added
- the code is generic, it is correct version and image id of newly created pod